Handle asyncio.CancelledError

anthony-symphony opened the pull request:
### Ticket
PLAT-10280

### Description
We currently do not handle asyncio.CancelledError. In certain situations this could cause the datafeed to keep reading. This fixes it by catching the CancelledError exception, stopping the datafeed and rethrowing it without recreating the datafeed.

anthony-symphony committed:
Handle asyncio.CancelledError

symphony-youri commented on sym_api_client_python/datafeed_event_service.py:
```diff
@@ -270,6 +272,10 @@ async def handle_datafeed_errors(self, thrown_exception):
             raise thrown_exception
+        except CancelledError as exc:
+            log.info("Cancel request received. Stopping datafeed...")
+            await self.deactivate_datafeed()
```
Do we want for the pending query to complete here? Or could it be faster to pass wait_for_handler_completions=False?

symphony-youness reviewed (dismissed):
Apparently they used a different way of running an async loop (which is not how we describe it in the python SDK's documentation) but it is one way to do it.
It explains the error we can't reproduce if we follow the SDK's documentation.

using: `asyncio.run(loop)`

I couldn't reproduce the hanging shell but there is indeed a non-handled `asyncio.CancelledError`

**There is no clear solution and it depends on how the keyboard interrupt is triggered:** https://github.com/ethereum/py-evm/issues/385#issuecomment-367611201

The PR does the handling of the exception it and is non-blocking for the documented way of running the async loop. 

- Here is the base issue raised on python git: https://github.com/python/asyncio/issues/341

anthony-symphony committed:
Moved where CancelledError Exception is handled
Needed for 3.8+ compatbility as CancelledError was changed from an Exception to a BaseException

symphony-elias reviewed (approved):
Looks good !

symphony-youri committed:
Merge remote-tracking branch 'upstream/master' into PLAT-10280
